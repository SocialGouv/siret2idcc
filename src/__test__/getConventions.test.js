const getConventions = require("../getConventions");

const tests = ["82161143100015", "81431448000017", "44858080300022"];

tests.forEach((input) => {
  test(`can get conventions for ${input}`, async () => {
    // timeout due to asynchronous data load
    await new Promise((resolve) => setTimeout(resolve, 500));
    expect(getConventions(input)).toMatchSnapshot();
  });
});
