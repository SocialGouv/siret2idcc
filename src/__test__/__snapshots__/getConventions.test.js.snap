// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can get conventions for 44858080300022 1`] = `Array []`;

exports[`can get conventions for 81431448000017 1`] = `Array []`;

exports[`can get conventions for 82161143100015 1`] = `
Array [
  Object {
    "active": true,
    "date_publi": "2005-01-01T00:00:00.000Z",
    "effectif": 63554,
    "etat": "VIGUEUR_ETEN",
    "id": "KALICONT000005635594",
    "mtime": 1562096111,
    "nature": "IDCC",
    "num": 1483,
    "shortTitle": "Commerce de détail de l'habillement et des articles textiles",
    "texte_de_base": "KALITEXT000005675211",
    "title": "Convention collective nationale du commerce de détail de l'habillement et des articles textiles du 25 novembre 1987, révisée par avenant du 17 juin 2004",
    "url": "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635594",
  },
  Object {
    "active": true,
    "date_publi": "1988-01-01T00:00:00.000Z",
    "effectif": 857061,
    "etat": "VIGUEUR_ETEN",
    "id": "KALICONT000005635173",
    "mtime": 1556652289,
    "nature": "IDCC",
    "num": 1486,
    "shortTitle": "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
    "synonymes": Array [
      "syntec",
    ],
    "texte_de_base": "KALITEXT000005679895",
    "title": "Convention collective nationale des bureaux d'études techniques, des cabinets d'ingénieurs-conseils et des sociétés de conseils du 15 décembre 1987. ",
    "url": "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635173",
  },
]
`;
